File: ios/marvel/Data/Models/DataResponse.swift
//
//  DataResponse.swift
//  marvel
//
//  Created by Vilmar Bispo Filho on 02/06/22.
//

import Foundation

struct DataResponse<T: Decodable>: Decodable {
    let offset: Int
    let limit: Int
    let total: Int
    let count: Int
    let results: [T]
    
    enum CodingKeys: String, CodingKey {
        case offset
        case limit
        case total
        case count
        case results
    }
}

File: android_app/data/src/main/java/br/com/data/model/DataResponse.kt
package br.com.data.model

import com.squareup.moshi.Json

data class DataResponse<T>(
   @Json(name = "offset")
   val offset: Int,
   @Json(name = "limit")
   val limit: Int,
   @Json(name = "total")
   val total: Int,
   @Json(name = "count")
   val count: Int,
   @Json(name = "results")
   val results: T
)
